persistence/context.go

```diff
@@ -47,9 +48,9 @@ func (p *PostgresContext) SetSavePoint() error {
 // RollbackToSavepoint triggers a rollback for the current pgx transaction and the underylying submodule stores.
 func (p *PostgresContext) RollbackToSavePoint() error {
 	ctx, _ := p.getCtxAndTx()
-	err := p.tx.Rollback(ctx)
-	p.stateTrees.Rollback()
-	return err
+	pgErr := p.tx.Rollback(ctx)
+	treesErr := p.stateTrees.Rollback()
+	return multierr.Combine(pgErr, treesErr)
```

Comment:
```suggestion
	return errors.Join(pgErr, treesErr)
```
Now we are in go 1.20 we can use this which works with `errors.Is()` https://pkg.go.dev/errors@master#Join

Reply:
oh, right! Good call, will update to match.

Reply:
If you haven't already done this, I think a separate PR just updating this across the board would be good. If you already have, ignore.